### PR TITLE
[release-4.12] remove k8s deps

### PR DESCRIPTION
**Pull request by ffromani**
targeted backport of commits from various PRs needed to be able to remove all the k8s deps from the branch 4.12

many PRs had layered/intertwined dep trees, so picking individual commits is the most practical way to go.

### CODE REVIEW

File: test/e2e/rte/rte_test.go [hunk 1]
```diff
@@ -280,17 +262,17 @@ var _ = ginkgo.Describe("with a running cluster with all the components", func()
 	})
 
 	ginkgo.It("[rte][podfingerprint] should expose the pod set fingerprint status on each worker", func() {
-		nropObj, err := nropcli.NUMAResourcesOperators().Get(context.TODO(), objectnames.DefaultNUMAResourcesOperatorCrName, metav1.GetOptions{})
-		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		nropObj := &nropv1.NUMAResourcesOperator{}
+		err := clients.Client.Get(context.TODO(), client.ObjectKey{Name: objectnames.DefaultNUMAResourcesOperatorCrName}, nropObj)
```
Comment (shajmakh) [inline test/e2e/rte/rte_test.go:259]: nit: missed error handling/reporting here

File: rte/main.go [hunk 2]
```diff
@@ -96,7 +96,11 @@ func main() {
 
 	// TODO: recycled flag (no big deal, but still)
 	cli = podexclude.NewFromLister(cli, parsedArgs.RTE.Debug, parsedArgs.LocalArgs.PodExcludes)
-	err = resourcetopologyexporter.Execute(cli, parsedArgs.NRTupdater, parsedArgs.Resourcemonitor, parsedArgs.RTE)
+	hnd := resourcemonitor.Handle{
+		PodResCli: cli,
+		K8SCli:    k8scli,
```
Comment (shajmakh) [inline rte/main.go:107]: I see k8cli is defined only in a later [commit](https://github.com/openshift-kni/numaresources-operator/pull/772/commits/6143f74d62b16860806c0216924e4de3cf0731d3#diff-fa3097e800fda70b00ec99071e568704f2584b6b7759c8f6149628535e90a8ceR85).

**Review by shajmakh — COMMENTED** [review]
Thanks for the PR. minor comments below. 
/lgtm 
/hold 
for others to review. feel free to unhold if as you see fit.

File: rte/main.go (hunk 2 above)
Comment (ffromani) [reply rte/main.go:107]: fixed the ordering, thanks!

File: test/e2e/rte/rte_test.go (hunk 1 above)
Comment (ffromani) [reply test/e2e/rte/rte_test.go:259]: thanks, fixed! even though is a backport, we need 4.12-specific changes anyway, so let's take the chance and add it.

**Comment by ffromani** [comment]
/hold cancel

tested locally with good results

**Pull request merged**